Fix flakey test

### Description

CI is failing randomly due to a flakey test in interactive-messages package.

```
1) SlackMessageAdapter
       #dispatch()
         when dispatching a message attachment action request
           should handle the callback returning a promise that fails before the timeout with a synchronous response:
     Error: Timeout of 50ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/home/runner/work/node-slack-sdk/node-slack-sdk/packages/interactive-messages/src/adapter.spec.js)
      at runNextTicks (internal/process/task_queues.js:58:5)
      at processTimers (internal/timers.js:494:9)




lerna ERR! npm run test stderr:
(node:3497) UnhandledPromiseRejectionWarning: Error: test error
    at delayed (/home/runner/work/node-slack-sdk/node-slack-sdk/packages/interactive-messages/test/helpers.js:96:13)
    at SlackMessageAdapter.<anonymous> (/home/runner/work/node-slack-sdk/node-slack-sdk/packages/interactive-messages/src/adapter.spec.js:653:18)
    at SlackMessageAdapter.dispatch (/home/runner/work/node-slack-sdk/node-slack-sdk/packages/interactive-messages/src/adapter.ts:95:78)
    at Context.<anonymous> (/home/runner/work/node-slack-sdk/node-slack-sdk/packages/interactive-messages/src/adapter.spec.js:655:41)
    at callFn (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runnable.js:387:21)
    at Test.Runnable.run (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runnable.js:379:7)
    at Runner.runTest (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:535:10)
    at /home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:653:12
    at next (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:447:14)
    at /home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:457:7
    at next (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:362:14)
    at /home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:420:7
    at done (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runnable.js:334:5)
    at callFn (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runnable.js:410:7)
    at Hook.Runnable.run (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runnable.js:379:7)
    at next (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:384:10)
    at Immediate.<anonymous> (/home/runner/work/node-slack-sdk/node-slack-sdk/node_modules/mocha/lib/runner.js:425:5)
    at processImmediate (internal/timers.js:461:21)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:3497) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:3497) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @slack/interactive-messages@2.0.0 test: `nyc mocha --config .mocharc.json src/*.spec.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @slack/interactive-messages@2.0.0 test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-01-26T01_08_00_936Z-debug.log

lerna ERR! npm run test exited 1 in '@slack/interactive-messages'
lerna WARN complete Waiting for 2 child processes to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @ test: `lerna run test`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @ test script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-01-26T01_08_00_994Z-debug.log
```

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [ ] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [x] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Packages:

Select all that apply:

- [x] `@slack/interactive-messages`

--- a/packages/interactive-messages/src/adapter.spec.js
+++ b/packages/interactive-messages/src/adapter.spec.js
@@ -662,7 +662,7 @@ describe('SlackMessageAdapter', function () {
         const timeout = this.adapter.syncResponseTimeout;
         this.timeout(timeout);
         this.adapter.action(requestPayload.callback_id, function () {
-          return delayed(timeout * 0.1, undefined, 'test error');
+          return delayed(0, undefined, 'test error');
         });
         dispatchResponse = this.adapter.dispatch(requestPayload);
         return assertResponseStatusAndMessage(dispatchResponse, 500);

--- a/packages/interactive-messages/test/helpers.js
+++ b/packages/interactive-messages/test/helpers.js
@@ -85,7 +85,8 @@ function createStreamRequest(signingSecret, ts, rawBody) {
 /**
  * Returns a Promise that resolves or rejects in approximately the specified amount of time with
  * the specified value or error reason.
- * @param {number} ms time in milliseconds in which to resolve or reject
+ * @param {number} ms time in milliseconds in which to resolve or reject. Zero is a special value which creates a
+ * Promise that resolves or rejects as soon as possible.
  * @param {*} value value used for resolve
  * @param {string} [rejectionReason] reason used for rejection
  * @returns {Promise<*>} a promise of the value type
@@ -96,14 +97,28 @@ function delayed(ms, value, rejectionReason) {
     error = new Error(rejectionReason);
   }
   return new Promise(function (resolve, reject) {
-    var id = setTimeout(function () {
-      clearTimeout(id);
-      if (error) {
-        reject(error);
-      } else {
-        resolve(value);
-      }
-    }, ms);
+    if (ms > 0) {
+      var id = setTimeout(function () {
+        clearTimeout(id);
+        if (error) {
+          reject(error);
+        } else {
+          resolve(value);
+        }
+      }, ms);
+    } else if (ms === 0) {
+      // setImmediate allows the returned Promise to resolve much sooner than setTimeout.
+      // See: https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#setimmediate-vs-settimeout
+      setImmediate(() => {
+        if (error) {
+          reject(error);
+        } else {
+          resolve(value);
+        }
+      });
+    } else {
+      reject(new Error('Cannot delay for a negative amount of time'));
+    }
   });
 }
 

